framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -792,23 +792,14 @@ public void initChecker() {
     protected int errsOnLastExit = 0;
 
     /**
-     * Type-check the code with Java specifications and then runs the Checker Rule Checking visitor
-     * on the processed source.
+     * Type-check the code using this checker's visitor.
      *
      * @see Processor#process(Set, RoundEnvironment)
      */
     @Override
     public void typeProcess(TypeElement e, TreePath p) {
-        if (e == null) {
-            messager.printMessage(
-                    javax.tools.Diagnostic.Kind.ERROR, "Refusing to process empty TypeElement");
-            return;
-        }
-        if (p == null) {
-            messager.printMessage(
-                    javax.tools.Diagnostic.Kind.ERROR,
-                    "Refusing to process empty TreePath in TypeElement: " + e);
-            return;
+        if (e == null || p == null) {
+            throw new BugInCF("SourceChecker.typeProcess: e=%s, p=%s", e, p);
```

Comment:
You can't use `BugInCF` up here, because it is outside of the corresponding try/catch that would do something useful with it. So in the unlikely event this would ever happen, an ugly stack trace will be shown.
The previous code produces separate understandable errors, which seems preferable.
How about undoing this change?